### PR TITLE
Add Scala version of StalinSort.

### PR DESCRIPTION
**Pull request by pamu**
- Added scala version of stalin sort. :)

### CODE REVIEW

File: scala/StalinSort.scala [hunk 1]
```diff
@@ -0,0 +1,15 @@
+object StalinSort extends App {
+
+  def stalinSort[A: Ordering](xs: List[A]): List[A] = xs match {
+    case Nil          => Nil
+    case x :: Nil     => List(x)
+    case x :: y :: zs =>
+      if (implicitly[Ordering[A]].lteq(x, y)) x :: stalinSort(y :: zs)
```
Comment (bbstilson) [inline scala/StalinSort.scala]: Beat me to the Scala version!! 😛 

For the comparison, why not:

```scala
case x :: y :: zs => if (x < y) then x :: stalinSort(y :: zs) else stalinSort(x :: zs)
```

Comment (pamu) [reply scala/StalinSort.scala]: You are right for a concrete implementation but, this one is polymorphic.

```

scala>   def stalinSort[A: Ordering](xs: List[A]): List[A] = xs match {
     |     case Nil => Nil
     |     case x :: Nil => List(x)
     |     case x :: y :: zs =>
     |       if(x < y) x :: stalinSort(y :: zs) else stalinSort(x :: zs)
     | }
<console>:15: error: value < is not a member of type parameter A
             if(x < y) x :: stalinSort(y :: zs) else stalinSort(x :: zs)
                  ^
```

Comment (bbstilson) [reply scala/StalinSort.scala]: Thanks 👍

**Comment by gustavo-depaula** [comment]
there is already a scala version (#39). if your approach is different from this one, you can open a new file or suggest a modification.

**Comment by pamu** [comment]
> there is already a scala version (#39). if your approach is different from this one, you can open a new file or suggest a modification.

Yes, this is an functional version of stalin sort. I moved it to different dir called `scala-fp`

**Comment by gustavo-depaula** [comment]
@pamu instead of a new directory, name your file StalinSortFunciontal, pls

**Comment by pamu** [comment]
> @pamu instead of a new directory, name your file StalinSortFunciontal or something like that, pls

Done. Thanks

**Pull request merged**